### PR TITLE
Quit when EOF is reached

### LINKED ISSUE #680: defmt-print: exit when stdin is closed
When the standard input is closed, `defmt-print` uses 100% CPU by reading 0 bytes from stdin in a loop. I think the most useful behaviour is to exit once all received logs are print. This allows to use e.g. `nc -d localhost 8765 | defmt-print -e <elf> &` and to have `defmt-print` exit when the server is restarted.

https://github.com/knurling-rs/defmt/blob/7dff53f6c7c423b0d2c0287687b6882b3cbc1f7b/print/src/main.rs#L70-L73

### PR DESCRIPTION
**Pull request by kellda**
Closes #680

### CODE REVIEW

**Comment by Urhengulas** [comment]
Thank you for the PR! Please add your change to the `CHANGELOG.md` file to make the CI pass.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,11 +7,13 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+- [#680]: defmt-print: exit when stdin is closed
```
Comment (Urhengulas) [inline CHANGELOG.md:10]: Please reference the PR and not the issue:
```suggestion
- [#682]: defmt-print: exit when stdin is closed
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,11 +7,13 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
 - [#681]: Make use of i/o locking being static since rust `1.61`.
 - [#679]: Add changelog enforcer
 - [#678]: Satisfy clippy
 
 [#681]: https://github.com/knurling-rs/defmt/pull/681
+[#680]: https://github.com/knurling-rs/defmt/issues/680
```
Comment (Urhengulas) [inline CHANGELOG.md:16]:
```suggestion
[#682]: https://github.com/knurling-rs/defmt/pull/682
```

File: print/src/main.rs [hunk 3]
```diff
@@ -69,6 +69,10 @@ fn main() -> anyhow::Result<()> {
     loop {
         // read from stdin and push it to the decoder
         let n = stdin.read(&mut buf)?;
+        // if 0 bytes where read, we reached EOF, so quit
+        if n == 0 {
+            break Ok(());
```
Comment (Urhengulas) [inline print/src/main.rs:74]: I see you already fixed that 😄 was just about to tell you

**Review by Urhengulas — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (kellda) [reply CHANGELOG.md]: I thought the issue would be more interesting to reference as it has more infos and context… I fixed it now.

**Pull request merged**